sormas-backend/src/main/java/de/symeda/sormas/backend/externalemail/AttachmentService.java

```diff
@@ -0,0 +1,204 @@
+    private final Map<String, PdfConverter> converters = Map.of(
+            DOCX_FILE_EXTENSION, new DocXConverter(),
+            IMAGE_FILE_EXTENSTIONS, new ImageConverter()
+    );
+    // @formatter:on
+
+    public Set<String> getAttachableFileExtensions() {
+        return Set.of((PDF_FILE_EXTENSION + "," + DOCX_FILE_EXTENSION + "," + IMAGE_FILE_EXTENSTIONS).split(","));
+    }
+
+    @NotNull
+    public Map<File, String> createEncryptedPdfs(List<de.symeda.sormas.backend.document.Document> sormasDocuments, String passowrd) {
+        return sormasDocuments.stream().map(d -> {
+            String fileName = d.getName();
+            String fileExtension = fileName.substring(fileName.lastIndexOf("."));
+
+            try {
+                final File encryptedPdf;
+                File document = documentStorageService.getFile(d.getStorageReference());
+                if (fileExtension.equals(PDF_FILE_EXTENSION)) {
+                    encryptedPdf = encryptPdf(document, passowrd);
+                } else {
+                    PdfConverter converter = getConverter(fileExtension);
+                    fileName = converter.getConvertedFileName(fileName);
+                    File converted = converter.convert(document);
+                    encryptedPdf = encryptPdf(converted, passowrd);
+                    converted.delete();
+                }
+
+                return new AbstractMap.SimpleEntry<>(encryptedPdf, fileName);
+            } catch (IOException e) {
+                // not really expected to happen
+                throw new RuntimeException(e);
+            }
+        }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private File encryptPdf(File pdf, String password) throws IOException {
+        PDDocument pdd = Loader.loadPDF(pdf);
+
+        AccessPermission ap = new AccessPermission();
+        StandardProtectionPolicy stpp = new StandardProtectionPolicy(password, password, ap);
+        stpp.setEncryptionKeyLength(128);
+        stpp.setPermissions(ap);
+        pdd.protect(stpp);
+
+        String encryptedFile = getTmpFilePathForConversion() + "_encrypted" + PDF_FILE_EXTENSION;
+        pdd.save(encryptedFile);
+        pdd.close();
+
+        return new File(encryptedFile);
+    }
+
+    private PdfConverter getConverter(String fileExtension) {
+        Optional<String> converterKey = converters.keySet().stream().filter(k -> k.contains(fileExtension)).findFirst();
+        if (converterKey.isEmpty()) {
+            throw new IllegalArgumentException("No converter found for file extension " + fileExtension);
+        }
+
+        return converters.get(converterKey.get());
+    }
+
+    private Path getTmpFilePathForConversion() {
+        Path path = Paths.get(configFacade.getTempFilesPath());
+        String fileName = TEMP_FILE_PREFIX + "_converted_" + DateHelper.formatDateForExport(new Date()) + "_" + RANDOM.nextInt(Integer.MAX_VALUE);
+
+        return path.resolve(fileName);
```

Comment:
That's a runtime exception, so no need to catch it. Will be caught and logged by a generic error handler. If it somehow happens it can be considered a system issue so it's ok that the user will see a generic server error message.

Reply:
I did not opened it and assumed that is a checked exception (same for XWPFConverterException). From Intellij the unhandled checked exceptions are prompted but I've checked from github.